apps/patient/src/components/ReadyText.tsx

```diff
@@ -0,0 +1,106 @@
+import dayjs, { Dayjs } from 'dayjs';
+import { Text } from '@chakra-ui/react';
+import { OrderFulfillment } from 'packages/sdk/dist/types';
+
+export type Maybe<T> = T | null;
+interface ReadyTextProps {
+  readyBy?: string;
+  readyByDay?: string;
+  isDeliveryPharmacy?: boolean;
+  fulfillment?: Maybe<OrderFulfillment> | undefined;
+}
+const ReadyText = ({ readyBy, readyByDay, isDeliveryPharmacy, fulfillment }: ReadyTextProps) => {
+  const showPatientDesiredreadyBy =
+    readyBy &&
+    readyByDay &&
+    !isDeliveryPharmacy &&
+    (!fulfillment || // No fulfillment means user came from pharmacy selection
+      fulfillment?.state === 'SENT');
+  const showEstimatedReadyAt =
+    fulfillment?.pharmacyEstimatedReadyAt &&
+    !isDeliveryPharmacy &&
+    (!fulfillment || // No fulfillment means user came from pharmacy selection
+      fulfillment?.state === 'RECEIVED');
+
+  return (
+    <>
+      {showPatientDesiredreadyBy && (
+        <PatientDesiredReadyBy readyBy={readyBy} readyByDay={readyByDay} />
+      )}
+      {showEstimatedReadyAt && (
+        <PharmacyEstimatedReadyAt pharmacyEstimatedReadyAt={fulfillment.pharmacyEstimatedReadyAt} />
+      )}
+    </>
+  );
+};
+
+const roundUpTo15MinInterval = (pharmacyEstimatedReadyAt: Date): Dayjs => {
+  const dayJS = dayjs(pharmacyEstimatedReadyAt);
```

Comment:
```suggestion
  const estimatedReadyAtAsDayJs = dayjs(pharmacyEstimatedReadyAt);
```

`dayJS` and `dayjs` are VERY close to each other